lib/finch.ex

```diff
@@ -107,12 +121,16 @@ defmodule Finch do
 
   @impl true
   def init(config) do
+    ets_name = ets_name(config.registry_name)
+    config = Map.put(config, :ets_name, ets_name)
+
     children = [
       {DynamicSupervisor, name: config.supervisor_name, strategy: :one_for_one},
       {Registry, [keys: :duplicate, name: config.registry_name, meta: [config: config]]},
       {PoolManager, config}
     ]
 
+    :ets.new(ets_name, [:public, :named_table])
```

Comment:
Currently investigating just using the table reference (and not naming the table) and if :write_concurrency & :read_concurrency have any significant impact.